fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/entity/FabricEntityType.java

```diff
@@ -0,0 +1,151 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.entity;
+
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.SpawnGroup;
+import net.minecraft.entity.SpawnRestriction;
+import net.minecraft.entity.attribute.DefaultAttributeContainer;
+import net.minecraft.entity.mob.MobEntity;
+import net.minecraft.world.Heightmap;
+
+import net.fabricmc.fabric.impl.object.builder.FabricEntityTypeImpl;
+
+/**
+ * General-purpose Fabric-provided extensions for {@link EntityType}.
+ */
+public interface FabricEntityType {
+	/**
+	 * General-purpose Fabric-provided extensions for {@link EntityType.Builder}.
+	 *
+	 * <p>Note: This interface is automatically implemented on {@link EntityType.Builder} via Mixin and interface injection.
+	 */
+	interface Builder<T extends Entity> {
+		/**
+		 * Sets whether the entity's velocity should always be updated.
+		 *
+		 * @param alwaysUpdateVelocity whether the entity's velocity should always be updated
+		 * @return this builder
+		 */
+		default EntityType.Builder<T> alwaysUpdateVelocity(boolean alwaysUpdateVelocity) {
+			throw new AssertionError("Implemented in Mixin");
+		}
+
+		/**
+		 * Build the entity type from the builder. Same as {@link EntityType.Builder#build(String)} but without an id.
+		 *
+		 * @return the entity type instance
+		 */
+		default EntityType<T> build() {
+			throw new AssertionError("Implemented in Mixin");
+		}
+
+		/**
+		 * Creates an entity type builder for a living entity.
+		 *
+		 * <p>This entity's spawn group will automatically be set to {@link SpawnGroup#MISC}.
+		 *
+		 * @param <T> the type of entity
+		 * @param livingBuilder a function to configure living entity specific properties
+		 *
+		 * @return a new living entity type builder
+		 */
+		static <T extends LivingEntity> EntityType.Builder<T> createLiving(UnaryOperator<Living<T>> livingBuilder) {
+			return FabricEntityTypeImpl.Builder.createLiving(livingBuilder);
+		}
+
+		/**
+		 * Creates an entity type builder for a living entity.
+		 *
+		 * <p>This entity's spawn group will automatically be set to {@link SpawnGroup#MISC}.
+		 *
+		 * @param <T> the type of entity
+		 *
+		 * @return a new living entity type builder
+		 */
+		static <T extends LivingEntity> EntityType.Builder<T> createLiving() {
+			return FabricEntityTypeImpl.Builder.createLiving(UnaryOperator.identity());
+		}
+
+		/**
+		 * Creates an entity type builder for a mob entity.
```

Comment:
Where do we set the constructor? `PlayerEntity` is the only one in vanilla where constructor is skipped. Otherwise client sync/biome spawn/etc wouldn't work.

Reply:
Ah, I see good point. The previous API you would set the factory via the builder methods, of course this cannot happen now.